samcli/commands/validate/lib/sam_template_validator.py

```diff
@@ -120,6 +122,23 @@ def _replace_local_codeuri(self):
                 if "DefinitionUri" in resource_dict:
                     SamTemplateValidator._update_to_s3_uri("DefinitionUri", resource_dict)
 
+    def _replace_local_image(self):
+        """
+        Adds fake ImageUri to AWS::Serverless::Functions that reference a local image using Metadata.
+        This ensures sam validate works without having to package the app or use ImageUri.
+        """
+        resources = self.sam_template.get("Resources", {})
+        for _, resource in resources.items():
+            resource_type = resource.get("Type")
+            properties = resource.get("Properties", {})
+
+            is_image_function = resource_type == AWS_SERVERLESS_FUNCTION and properties.get("PackageType") == IMAGE
```

Comment:
Sorry, I missed this before, we need to consider AWS_LAMBDA_FUNCTION too?

Reply:
Not AFAIK since `AWS::Lambda::Function` doesn't have a notion of local images (and any pure CFN resources should pass validation).

Reply:
agree on pure CFN, however we do still build Image based AWS::Lambda::Function though IIRC.

Reply:
`sam validate` already works with any `AWS::Lambda::Function` such as:

```yaml
AWSTemplateFormatVersion: '2010-09-09'
Transform: AWS::Serverless-2016-10-31
Resources:
  Function:
    Type: AWS::Lambda::Function
    Properties: Something
```